python/cudf/udf_cpp/shim.cu

```diff
@@ -643,9 +643,8 @@ __device__ double BlockCorr(T* const lhs_ptr, T* const rhs_ptr, int64_t size)
 {
   auto numerator   = BlockCoVar(lhs_ptr, rhs_ptr, size);
   auto denominator = BlockStd(lhs_ptr, size) * BlockStd<T>(rhs_ptr, size);
-
   if (denominator == 0.0) {
-    return 0.0;
+    return std::numeric_limits<double>::quiet_NaN();
```

Comment:
Can we have a pytest added to test this?

Reply:
Added a simple test in https://github.com/rapidsai/cudf/pull/13884/commits/26ad24a7530563f553bbff3c297e4794aa3fee53. More tests pending, including one which covers this case, at https://github.com/rapidsai/cudf/pull/13854.